azure-kusto-data/azure/kusto/data/security.py

```diff
@@ -47,8 +47,10 @@ def _get_azure_cli_auth_token() -> dict:
 
             # TODO: not sure I should take the first
             return data[0]
-        except Exception:
-            pass
+        except Exception as e:
+            raise KustoAuthenticationError(
+                AuthenticationMethod.az_cli_profile, e, message="Azure cli token was not found. Please run 'az login' to setup account."
```

Comment:
This message is a bit specific for such a broad exception catch.
Would it fail here if az is not installed, not in env?
I'd consider a more broad message or more granular checks and messages.

Reply:
There are too many cases that can cause an exception here (file not found, empty file, wrong content, json parsing etc) - all of them should be solved by logging in to az cli.
We wrap the original exception, not replacing it, so this message its just a suggestion.

Reply:
Sure